plugin/plaintextmodel.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2019 Jolla Ltd.
+ * Contact: Andrew den Exter <andrew.den.exter@jolla.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#ifndef PLAINTEXTMODEL_H
+#define PLAINTEXTMODEL_H
+
+#include <QAbstractItemModel>
+#include <QCache>
+#include <QExplicitlySharedDataPointer>
+#include <QFile>
+#include <QTextStream>
+#include <QUrl>
+
+#include <vector>
+
+class PlainTextModel : public QAbstractItemModel
+{
+    Q_OBJECT
+    Q_PROPERTY(QUrl source READ source WRITE setSource NOTIFY sourceChanged)
+    Q_PROPERTY(Status status READ status NOTIFY statusChanged)
+    Q_PROPERTY(int count READ rowCount NOTIFY countChanged)
```

Comment:
Quite a minor thing, but ponder if naming like lineCount, or lines more emphasized in general, would be clearer.

Reply:
Changed to 'lineCount'. 'lines' is ambiguous, it could return the number of lines or return a list of them.